README.md

```diff
@@ -467,6 +467,20 @@ stacks:
       template: my-stack.rb
 ```
 
+### Loading SparklePacks
+
+[SparklePacks](http://www.sparkleformation.io/docs/sparkle_formation/sparkle-packs.html) can be pre-loaded using compiler options. This requires the name of a rubygem to `require` followed by the name of the SparklePack, which is usually the same name as the Gem.
+
+```yaml
+stacks:
+  us-east-1
+    my-stack:
+      template: my-stack-with-dynamic.rb
+      compiler_options:
+        sparkle_pack:
+          vpc-sparkle-pack: vpc-sparkle-pack
```

Comment:
This line is a little confusing as well. I can see it's `gemname: packname` but aren't they the same thing...
Can you get rid of the `packname`

Reply:
I don't know, but SparkleFormation supports it so I'm inclined to maintain that interface unless there's some compelling reason not to.

Reply:
When you traditionally loaded like this

```
require 'my_sparkle_pack'

custom_pack = SparkleFormation::SparklePack.new(:name => 'my_sparkle_pack')
sfn = SparkleFormation.new(:my_template) do
   dynamic!(:my_dynamic)
end
sfn.sparkle.add_sparkle(custom_pack)
``` 
They actually needed to be the same to work

Reply:
From what I understand it's entirely possible to load it like this:

```ruby
require 'my_sparkle_pack'

custom_pack = SparkleFormation::SparklePack.new(:name => 'wow_dude_a_sparkle_pack')
```

Where `my_sparkle_pack` registers itself like so:

```ruby
SparkleFormation::SparklePack.register!('wow_dude_a_sparkle_pack')
```

Reply:
@patrobinson that's undocumented, you have to physically look at the code to see that. Why would you make it hard to consume by naming the pack different to the gem? Seems an odd thing to do to me.

If you look at the how they tell you to search for sparkle-pack gems [here]( http://www.sparkleformation.io/docs/sfn/sparkle-packs.html) and look at gems from [here](https://rubygems.org/search?query=sparkle-pack)

No-one does that has implemented a pack has done that. They've all followed the doco from [here](http://www.sparkleformation.io/docs/sparkle_formation/sparkle-packs.html#distribution)  

https://github.com/reverseskate/sparkle-pack-cfn-init-chef/blob/master/lib/sparkle-pack-cfn-init-chef.rb
https://github.com/timurb/sparkle-pack-shell-helpers/blob/master/lib/sparkle-pack-shell-helpers.rb
https://github.com/reverseskate/sparkle-pack-aws-availability-zones/blob/master/lib/sparkle-pack-aws-availability-zones.rb
https://github.com/reverseskate/sparkle-pack-aws-amis/blob/develop/lib/sparkle-pack-aws-amis.rb
https://github.com/carnivore-rb/sparkle-pack-jackal-cfn/blob/master/lib/sparkle-pack-jackal-cfn.rb
https://github.com/timurb/sparkle-pack-rails-helpers/blob/master/lib/sparkle-pack-rails-helpers.rb
https://github.com/reverseskate/sparkle-pack-cfn-init-chef/blob/master/lib/sparkle-pack-cfn-init-chef.rb

I wouldn't cater for an undocumented edge case personally. I'd make it easy to use and understand.

Reply:
It's a supported API but whatever, done in 00e8993